tools/wptserve/wptserve/handlers.py

```diff
@@ -237,7 +237,7 @@ def __call__(self, request, response):
         try:
             environ = {"__file__": path}
             sys.path.insert(0, os.path.dirname(path))
-            execfile(path, environ, environ)
+            exec(compile(open(path).read(), path, 'exec'), environ, environ)
```

Comment:
Hmm.

This will pass the contents of `path` as a unicode str in Py3 (decoded based on the default system encoding) and a bytes str in Py2 (does `execfile` just match `compile` and treat it as Latin-1 then?). What's our current behaviour here? Do we pay attention to `coding` declarations?

It also will make us inherit `__future__` imports from this file, which is a bit weird (I think all we have to do to there is `dont_inherit=False` on the `compile` call?). What's our current behaviour here?

IMO, we should be following `coding` and `__future__` imports in the handler files, though obviously if we currently don't then we shouldn't block this PR on that.

That all said, I'd strongly much prefer to explicitly close the file object (given this makes a significant behavioural difference on other Python impls, most obviously PyPy), probably using a context manager for it.